syntax = "proto3";

package transaction_verification;

service TransactionVerificationService {
    rpc VerifyTransaction (VerificationRequest) returns (VerificationResponse);
}

message VerificationRequest {
    repeated ItemData items = 1;
    CreditCardData creditCard = 2;
}

message VerificationResponse {
    bool success = 1;
}

message ItemData {
    string name = 1;
    string quantity = 2;
}

message CreditCardData {
    string number = 1;
    string expirationDate = 2;
    string cvv = 3;
}